### ISSUE TITLE
CakePHP 3.0.0RC2 when baking to a plugin, bake controller doesn't find model's associations.

### ISSUE THREAD

**Issue by djamesfar** [cakephp/bake#57]
To reproduce:
bake a plugin:
```
bin\cake bake plugin Pluggy
```
bake a model in the plugin, where the table has belongsTo or hasMany associations (ie Roles):
```
bin\cake bake model --plugin Pluggy --no-test --force roles
bin\cake bake model --plugin Pluggy --no-test --force users
```
now bake the UsersController:
```
bin\cake bake controller --plugin Pluggy --no-test --force users
```
Check the UsersController::add() method and you won't see the find for the associated belongsTo table:
```
$roles = $this->Users->Roles->find('list',...etc)     <--- *this is missing*
```
so it doesn't set(compact('roles'..  for the view.

If you take out the _--plugin Pluggy_,  it will work correctly, creating the controller in _src/controller_, of course, not in _plugins/Pluggy/src/controller_.

I dug around a little in the add.ctp element template used by the controller template and inserted some debug logging after the $associations array is populated using Bake->aliasExtractor (around line 36), and it shows the array empty when baking to a plugin, filled if not.  I didn't dig into the source for this aliasExtractor, but somewhere I expect a _$plugin_ variable got left out.
DJ

**Comment by imagra** [comment]
Just stepped on the same rake. And can confirm that baking the same tables for App (without --plugin parameter) works well. Also as Djamesfar found that shit happens somewhere before function aliasExtractor – the function gets empty list of associations.

**Comment by djamesfar** [comment]
@imagra, since a fix for this looks like it keeps getting pushed down the road, I created a somewhat hairy workaround: we have an app running as a plugin with over 120 tables in the model.  I bake everything but the controller to the plugin, but the controller I bake without the --plugin option so the output goes to /src/Controller.  I wrote a script that modifies the 'uses' statement to include the plugin namespace in each controller file and then copy it to the /plugins/<plugin>/src/Controller directory.

**Comment by markstory** [comment]
@djamesfar Is this still an issue. I tried to reproduce based on your steps above (which were quite detailed thank you) and couldn't. The controller that was generated included the find(list) calls as it should.

**Comment by imagra** [comment]
Checked. Can be reproduced on clean install:
```
CREATE DATABASE IF NOT EXISTS `cake-test3` DEFAULT CHARACTER SET utf32 COLLATE utf32_unicode_ci;
USE `cake-test3`;
DROP TABLE IF EXISTS `roles`;
CREATE TABLE `roles` (
  `id` char(36) COLLATE utf8_unicode_ci NOT NULL,
  `name` varchar(2048) COLLATE utf8_unicode_ci NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
DROP TABLE IF EXISTS `users`;
CREATE TABLE `users` (
  `id` char(36) COLLATE utf8_unicode_ci NOT NULL,
  `name` varchar(255) COLLATE utf8_unicode_ci DEFAULT NULL,
  `role_id` char(36) COLLATE utf8_unicode_ci NOT NULL
) ENGINE=InnoDB DEFAULT CHARSET=utf8 COLLATE=utf8_unicode_ci;
ALTER TABLE `roles`
  ADD PRIMARY KEY (`id`);
ALTER TABLE `users`
  ADD PRIMARY KEY (`id`),
  ADD KEY `role_id` (`role_id`);
```
```
composer create-project --prefer-dist cakephp/app test3
cd ./test3
bin/cake bake plugin Guard
bin/cake bake all  Roles  --plugin Guard
bin/cake bake all  Users  --plugin Guard
```
By opening http://cake-test3.local/guard/users/add you can see error
PHP 5.6.7

**Comment by markstory** [comment]
Thanks @imagra I'll try again with these steps tonight.

**Comment by markstory** [comment]
Fixed with #83

**Comment by djamesfar** [comment]
Perfect, Mark.  Thanks.
